src/python-imports-search.ts

```diff
@@ -1,67 +1,91 @@
+import { from, Subscription } from 'rxjs'
+import { filter, map, switchMap } from 'rxjs/operators'
 import * as sourcegraph from 'sourcegraph'
+import { resolveSettings, Settings } from './settings'
+
+const decorationType = sourcegraph.app.createDecorationType && sourcegraph.app.createDecorationType()
+const settings = resolveSettings(sourcegraph.configuration.get<Settings>().value)
 
 export function activate(): void {
-   sourcegraph.search.registerQueryTransformer({
-       transformQuery: (query: string) => {
-           const pyImportsRegex = /\bpy.imports:([^\s]*)/i
-           if (query.match(pyImportsRegex)) {
-               const pyImportsFilter = query.match(pyImportsRegex)
-               const pyPkg = pyImportsFilter && pyImportsFilter.length >= 1 ? pyImportsFilter[1] : ''
-               const singleImport = '^import\\s' + pyPkg + '$'
-               const namedImport = '^import\\s' + pyPkg + '\\sas\\s[^\\s]*'
-               const scopedImport = '^from\\s(.*)\\simport\\s(.*)' + pyPkg
-               return query.replace(pyImportsRegex  , `(${singleImport}|${namedImport}|${scopedImport}) lang:python `)
-           }
-           return query
-        }
-   })
+    const subscriptions = new Subscription()
+
+    subscriptions.add(
+        sourcegraph.search.registerQueryTransformer({
+            transformQuery: (query: string) => {
+                const pyImportsRegex = /\bpy.imports:([^\s]*)/i
+                if (query.match(pyImportsRegex)) {
+                    const pyImportsFilter = query.match(pyImportsRegex)
+                    const pyPkg = pyImportsFilter && pyImportsFilter.length >= 1 ? pyImportsFilter[1] : ''
+                    const singleImport = '^import\\s' + pyPkg + '$'
+                    const namedImport = '^import\\s' + pyPkg + '\\sas\\s[^\\s]*'
+                    const scopedImport = '^from\\s(.*)\\simport\\s(.*)' + pyPkg
+
+                    return query.replace(
+                        pyImportsRegex,
+                        `(${singleImport}|${namedImport}|${scopedImport}) lang:python patternType:regexp `
+                    )
+                }
+                return query
+            },
+        })
+    )
+
+    const editorsChanges = sourcegraph.app.activeWindowChanges
+        ? from(sourcegraph.app.activeWindowChanges).pipe(
+              filter(
+                  (activeWindow): activeWindow is Exclude<typeof activeWindow, undefined> => activeWindow !== undefined
+              ),
+              switchMap(activeWindow =>
+                  from(activeWindow.activeViewComponentChanges).pipe(map(() => activeWindow.visibleViewComponents))
+              )
+          )
+        : from(sourcegraph.workspace.openedTextDocuments).pipe(
+              map(() => (sourcegraph.app.activeWindow && sourcegraph.app.activeWindow.visibleViewComponents) || [])
+          )
+
+    subscriptions.add(
+        editorsChanges.subscribe(codeEditors => {
+            const codeEditor = codeEditors[0]
```

Comment:
Wouldn't you want to decorate _all_ editors when editorsChanges emits?

Reply:
Is this to accommodate diffs/PRs?

Reply:
Yes.